pkg/daemon/update_test.go

```diff
@@ -225,6 +225,26 @@ func TestMachineConfigDiff(t *testing.T) {
 	assert.False(t, diff.isEmpty())
 	assert.True(t, diff.osUpdate)
 
+	osImageURL := "quay.io/example/foo@sha256:b5bb9d8014a0f9b1d61e21e796d78dccdf1352f23cd32812f4850b878ae4944c"
+	oldConfig.Spec.OSImageURL = "docker://" + osImageURL
+	newConfig.Spec.OSImageURL = osImageURL
+	diff, err = newMachineConfigDiff(oldConfig, newConfig)
+	assert.Nil(t, err)
+	assert.True(t, diff.isEmpty())
+
+	oldConfig.Spec.OSImageURL = "registry:" + osImageURL
+	newConfig.Spec.OSImageURL = osImageURL
+	diff, err = newMachineConfigDiff(oldConfig, newConfig)
+	assert.Nil(t, err)
+	assert.True(t, diff.isEmpty())
+
+	oldConfig.Spec.OSImageURL = "registryy:" + osImageURL
+	newConfig.Spec.OSImageURL = osImageURL
+	diff, err = newMachineConfigDiff(oldConfig, newConfig)
+	assert.Nil(t, err)
+	assert.False(t, diff.isEmpty())
+	assert.True(t, diff.osUpdate)
+
```

Comment:
Can we also add a test case where oldConfig has empty OSImageURL and newConfig has a valid OSImageURL. something like oldConfig.Spec.OSImageURL = " "  and newConfig.Spec.OSImageURL = ""quay.io/example/foo@sha256:b5bb9d8014a0f9b1d61e21e796d78dccdf1352f23cd32812f4850b878ae4944c" .

This usually is the case today during initial node bootstrap for a cluster.

Reply:
Doesn't this already exist on line 222?

Reply:
ah yes, should have looked at existing test cases.